components/ai_chat/content/browser/bookmarks_service.h

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#ifndef BRAVE_COMPONENTS_AI_CHAT_CONTENT_BROWSER_BOOKMARKS_SERVICE_H_
+#define BRAVE_COMPONENTS_AI_CHAT_CONTENT_BROWSER_BOOKMARKS_SERVICE_H_
+
+#include "base/memory/raw_ptr.h"
+#include "base/scoped_observation.h"
+#include "brave/components/ai_chat/core/common/mojom/bookmarks.mojom.h"
+#include "components/bookmarks/browser/bookmark_model.h"
+#include "components/bookmarks/browser/bookmark_model_observer.h"
+#include "mojo/public/cpp/bindings/pending_receiver.h"
+#include "mojo/public/cpp/bindings/pending_remote.h"
+#include "mojo/public/cpp/bindings/receiver.h"
+#include "mojo/public/cpp/bindings/remote_set.h"
+
+namespace bookmarks {
+class BookmarkModel;
+}
+
+namespace ai_chat {
+
+class BookmarksService : public mojom::BookmarksService,
```

Comment:
why do we have a BookmarksService inside aichat? This class definitely needs some explanation in comments, but I don't understand why this would be necessary when there is already a BookmarksService and various observers.